src/show.jl

```diff
@@ -189,13 +210,26 @@ function _tree_children(parent::Union{Attribute, Datatype}, attributes::Bool)
 end
 
 function _show_tree(io::IO, obj::Union{File,Group,Dataset,Datatype,Attributes,Attribute}, indent::String="";
-                    attributes::Bool = true)
+                    attributes::Bool = true, depth::Int = 1)
     isempty(indent) && _tree_head(io, obj)
     !isvalid(obj) && return
```

Comment:
unrelated, but this reads better 
```suggestion
    isvalid(obj) || return
```
I think usually with `&&` or `||`, the  first  check should evaluate to `true`  for  action.